python/beeai_framework/utils/cloneable.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2025 IBM Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Protocol, TypeVar, runtime_checkable
+
+T = TypeVar("T", bound="Cloneable")
+
+
+@runtime_checkable
+class Cloneable(Protocol):
```

Comment:
Should we be exporting this in `__init__.py` or is it supposed to be internal only? @Tomas2D

Reply:
It's more like an internal thing.